components/camera/videosource/helpers.go

```diff
@@ -15,9 +16,38 @@ import (
 	"go.viam.com/rdk/utils"
 )
 
-func decodeImage(imgData []byte) (image.Image, error) {
-	img, _, err := image.Decode(bytes.NewBuffer(imgData))
-	return img, err
+func checkMimeType(ctx context.Context, data []byte) (string, error) {
+	detectedMimeType := http.DetectContentType(data)
+	requestedMime := gostream.MIMETypeHint(ctx, "")
+	actualMime, _ := utils.CheckLazyMIMEType(requestedMime)
+	if actualMime == utils.MimeTypeRawRGBA {
+		if detectedMimeType != utils.MimeTypeDefault {
+			return "", errors.Errorf(
+				"attempted to decode data using %s format as raw rgba data, "+
+					" raw rgba data must be encoded with our custom header",
+				detectedMimeType,
+			)
+		}
+		formatMatch := bytes.Compare(data[:4], rimage.RGBABitmapMagicNumber)
+		if formatMatch != 0 {
+			return "", errors.New(
+				"attempted to decode raw rgba data, but data was not encoded with the expected header format")
+		}
+	} else if actualMime != "" && actualMime != detectedMimeType {
+		return "", errors.Errorf(
+			"mime type requested (%q) for lazy decode not returned (got %q)",
```

Comment:
We haven't check that a lazy mime type was requested here.

Reply:
in the event of a lazy mime type being requested, isn't `actualMime` here representing the non-lazy type being requested (e.g. if `requstedMime` is "image/png+lazy", `actualMime` would be "image/png")?

Reply:
I'm saying that we need to check whether a lazy mime type was requested before returning this error .

Reply:
I put lazy decode in the error statement, which was a mistake. My intention was to do this check regardless of whether the original type requested a LazyEncodedImage. Is that the behavior we want though?

Reply:
oh should I change DecodeImage back then? Just calling image.Decode will use the MIME type in the header of the data

Reply:
I think you can leave DecodeImage as is, but pass it the detected mime type.